cms/djangoapps/contentstore/tests/test_courseware_index.py

```diff
@@ -551,15 +571,227 @@ def setUp(self):
             publish_item=False,
         )
 
+        self.library = LibraryFactory.create()
+
+        self.library_block1 = ItemFactory.create(
+            parent_location=self.library.location,
+            category="html",
+            display_name="Html Content",
+            publish_item=False,
+        )
+
+        self.library_block2 = ItemFactory.create(
+            parent_location=self.library.location,
+            category="html",
+            display_name="Html Content 2",
+            publish_item=False,
+        )
+
     def test_task_indexing_course(self):
         """ Making sure that the receiver correctly fires off the task when invoked by signal """
-        searcher = SearchEngine.get_search_engine(INDEX_NAME)
+        searcher = SearchEngine.get_search_engine(CoursewareSearchIndexer.INDEX_NAME)
         response = searcher.search(field_dictionary={"course": unicode(self.course.id)})
         self.assertEqual(response["total"], 0)
 
-        #update_search_index(unicode(self.course.id), datetime.now(UTC).isoformat())
+        # update_search_index(unicode(self.course.id), datetime.now(UTC).isoformat())
         listen_for_course_publish(self, self.course.id)
 
         # Note that this test will only succeed if celery is working in inline mode
         response = searcher.search(field_dictionary={"course": unicode(self.course.id)})
         self.assertEqual(response["total"], 3)
+
+    def test_task_library_update(self):
+        """ Making sure that the receiver correctly fires off the task when invoked by signal """
+        searcher = SearchEngine.get_search_engine(LibrarySearchIndexer.INDEX_NAME)
+        library_search_key = unicode(normalize_key_for_search(self.library.location.library_key))
+        response = searcher.search(field_dictionary={"library": library_search_key})
+        self.assertEqual(response["total"], 0)
+
+        # update_search_index(unicode(self.library.location.library_key), datetime.now(UTC).isoformat())
```

Comment:
@Kelketek I'll keep this comment just to avoid removing the "source" one a couple of lines above (the one @martynjames was talking about) - that commit was added in upstream master so there must be some value.